### PR TITLE
base: Check downloaded jfrog sha256 sum

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
For this we also install the handy tool jq to query json data.

### CODE REVIEW

**Comment by leandro-lucarella-sociomantic** [comment]
Waiting on https://github.com/JFrogDev/jfrog-cli-go/issues/62

**Review by mihails-strasuns-sociomantic — APPROVED** [review]

**Comment by leandro-lucarella-sociomantic** [comment]
OK, that URL needs authentication, so I was just getting a json response. Updated.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

File: docker/base [hunk 1]
```diff
@@ -55,5 +55,32 @@ gem install $(gem_ver travis "$VERSION_TRAVIS")
...
+			select(.path == \"$ver/jfrog-cli-linux-amd64/jfrog\") |
+			.sha256")
+
+	# Get the file
+	(set -x; curl -#LkX GET \
+		"https://dl.bintray.com/jfrog/jfrog-cli-go/$ver/jfrog-cli-linux-amd64/jfrog" \
+		> "$dst")
+
+	# Check file's sha256 sum
+	local sha256new="$(sha256sum "$dst" | cut -c1-64)"
```
Comment (nemanja-boric-sociomantic) [inline docker/base:79]: You could just use `sha256new` for this, though:

```
echo "$sha256 $dst" | sha256sum --quiet "$dst"
```

Comment (nemanja-boric-sociomantic) [reply docker/base:79]: Nevermind, this seems to fail though. I remember seeing it used like this.

Comment (nemanja-boric-sociomantic) [reply docker/base:79]: Ok, this is the right one:

```
echo "$sha256  $dst" | sha256sum -c --quiet
```

Comment (leandro-lucarella-sociomantic) [reply docker/base:79]: True (it's actually more like `echo "$sha256  $dst" | sha256sum -c -`) but this gives a better debug output. The check will say just if it failed or succeeded but if it fails it won't show the gotten and expected sums. So I prefer slightly the more verbose version.

Comment (nemanja-boric-sociomantic) [reply docker/base:79]: Yeah, fair enough, I have no issue with the current approach.

**Comment by leandro-lucarella-sociomantic** [comment]
ouch, this was broken. I don't know why travis didn't test

**Comment by leandro-lucarella-sociomantic** [comment]
In general if I'm assigned to the PR is because I'm not 100% sure about it and I plan to merge it myself after it is approved, I knew it was broken for a while now, sorry that I didn't say so explicitly! :grimacing:

**Comment by nemanja-boric-sociomantic** [comment]
Bah, I didn't see that you were assigned. I thought Travis _did_ test it and I couldn't spot anything wrong, sorry for merging. TBH, I really prefer not to merge these shell scripts, for this reason ;-)

**Pull request merged**